### PR TITLE
feat: [PL-28022]: Added "+User group" in select user group modal

### PR DESCRIPTION
**Pull request by avinashmadhwani02**
#### Screenshots

Screenshot 2022-11-10 at 7 45 32 PM

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/10-common/strings/strings.en.yaml [hunk 1]
```diff
@@ -1098,3 +1098,9 @@ resourceIdentifier: Resource Identifier
 configureNav: Or configure your nav!
 pipelineExecution: Pipeline Executions
 platformServiceVersions: Platform Service Versions
+notifications:
```
Comment (abhinavrastogi-harness) [inline src/modules/10-common/strings/strings.en.yaml:1101]: why is this moved to common? why not in `21-notifications` only?

File: src/modules/20-rbac/pages/UserGroupDetails/UserGroupDetails.tsx [hunk 2]
```diff
@@ -65,7 +65,7 @@ const UserGroupDetails: React.FC = () => {
   })
 
   const { openUserGroupModal } = useUserGroupModal({
-    onSuccess: refetch
+    onSuccess: () => refetch()
```
Comment (abhinavrastogi-harness) [inline src/modules/20-rbac/pages/UserGroupDetails/UserGroupDetails.tsx:68]: why is this change required?
are we sure this will not call `refetch()` prematurely when the component renders?

**Review by kapilkumawat86 — APPROVED** [review]

File: src/modules/20-rbac/pages/UserGroupDetails/UserGroupDetails.tsx (hunk 2 above)
Comment (avinashmadhwani02) [reply src/modules/20-rbac/pages/UserGroupDetails/UserGroupDetails.tsx:68]: There was a mismatch in the type of `refetch` and what onSuccess expects. Made this change to fix that. 

This will not impact the way refetch was getting called earlier.

**Review by harshilbits — APPROVED** [review]

**Review by vardanbansal-harness — APPROVED** [review]
